modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/OpenAPIResourceTest.java

```diff
@@ -193,59 +193,70 @@ public void testGetOpenAPI() throws Exception {
 
 	@Test
 	public void testGetOpenAPIInDifferentCompany() throws Exception {
-		JSONObject jsonObject = HTTPTestUtil.invokeToJSONObject(
-			JSONUtil.put(
-				"domain", "able.com"
-			).put(
-				"portalInstanceId", "able.com"
-			).put(
-				"virtualHost", "www.able.com"
-			).toString(),
-			"headless-portal-instances/v1.0/portal-instances",
-			Http.Method.POST);
-
-		HTTPTestUtil.customize(
-		).withBaseURL(
-			"http://www.able.com:8080"
-		).withCredentials(
-			"test@able.com", PropsValues.DEFAULT_ADMIN_PASSWORD
-		).apply(
-			() -> {
-				User user = UserTestUtil.addUser(
-					_companyLocalService.getCompany(
-						jsonObject.getLong("companyId")));
-
-				ObjectDefinition companyObjectDefinition =
-					ObjectDefinitionTestUtil.publishObjectDefinition(
-						"Object1",
-						Collections.singletonList(
-							ObjectFieldUtil.createObjectField(
-								ObjectFieldConstants.BUSINESS_TYPE_TEXT,
-								ObjectFieldConstants.DB_TYPE_STRING, true, true,
-								null, "field", "field", false)),
-						ObjectDefinitionConstants.SCOPE_COMPANY,
-						user.getUserId());
-
-				Assert.assertEquals(
-					200,
-					HTTPTestUtil.invokeToHttpCode(
-						null, companyObjectDefinition.getRESTContextPath(),
-						Http.Method.GET));
-
-				JSONObject openAPIJSONObject = HTTPTestUtil.invokeToJSONObject(
-					null, "/openapi", Http.Method.GET);
-
-				JSONArray jsonArray = openAPIJSONObject.getJSONArray(
-					companyObjectDefinition.getRESTContextPath());
-
-				Assert.assertEquals(1, jsonArray.length());
-				Assert.assertEquals(
-					"http://www.able.com:8080/o" +
+		long companyId = 0;
+
+		try {
+			JSONObject jsonObject = HTTPTestUtil.invokeToJSONObject(
```

Comment:
This code should be out of the try block. You will just delete the companyId once the company has been created

Reply:
Why is better? If the try fails to create the company, the companyId will be 0 and the delete process won't be triggered.

Reply:
Because in that case, you would be executing code that is not necessary (the if clause).

If you finally want to release what you created, it makes sense to start the try block after creating the resource, not before.